src/app/requests.service.ts

```diff
@@ -133,11 +203,46 @@ export class RequestsService {
     this.currentWeatherData.snow = request.snow?.['1h'];    
   }
 
+  makeForecastWeatherObject(request : forecastWeatherObject){
+    this.forcastWearherData.location = request.city.name;
+    this.forcastWearherData.forecastArr = []
+    for(let i = 0; i < request.list.length; i++){
+      let datas = {} as weatherDataObj
+      datas.time = new Date(request.list[i].dt_txt).toLocaleTimeString(this.lang, {
+        weekday: 'long',
+        // month: 'short',
+        // day: 'numeric',
+        hour: "numeric",
+        dayPeriod:"short"
+      })+'h';
+      datas.weather_icon = request.list[i].weather[0].icon;
+      datas.weather = request.list[i].weather[0].main;
+      datas.temperature = Math.round(request.list[i].main.temp);
+      datas.windSpeed = Math.round(request.list[i].wind.speed);
+      datas.cloudness = request.list[i].clouds.all;
+      datas.pressure = request.list[i].main.pressure;
+      datas.humidity = request.list[i].main.humidity;
+      datas.rain = request.list[i].rain?.['3h'];
+      datas.snow = request.list[i].snow?.['3h'];
+      this.forcastWearherData.forecastArr.push(datas);
+    }
+    console.log(this.forcastWearherData)
+  }
+
   formatedDate(unix_timestamp : number){
     const date = new Date(unix_timestamp * 1000);
     const hours = date.getHours();
     const minutes = date.getMinutes();
     return hours + ':' + minutes;
   }
+
+  time(date: string){
+    const month = new Date(date).getMonth() + 1;
+    const day = new Date(date).getDate();
+    const weekday = ["Sun","Mon","Tues","Wed","Thur","Fri","Sat"];
+    const dOfWeek = weekday[new Date(date).getDay()];
```

Comment:
this method is not in use. Now you use toLocaleTimeString to format dates. This can be deleted.